### ISSUE TITLE
ISC004: concatenating more than two lines?

### ISSUE THREAD

**Issue by hjoliver** [flake8-implicit-str-concat/flake8-implicit-str-concat#82]
Adding a line to your intentional concatenation example:
```Python
facts = (
    "Lobsters have blue blood.",
    "The liver is the only human organ that can fully regenerate itself.",
    (
        "Clarinets are made almost entirely out of wood from the mpingo tree."
        "In 1971, astronaut Alan Shepard played golf on the moon."
        "Or did he?"  # NEW LINE
    ),
)
```
This fails:
```console
$ flake8 test.py
test.py:5:79: ISC004 unparenthesized implicit string concatenation in collection (missing comma?)
test.py:6:67: ISC004 unparenthesized implicit string concatenation in collection (missing comma?)
```

Delving a little deeper, this does not trigger the warning (although the concatenation could be unintentional):
```Python
facts = (
    "Clarinets are made almost entirely out of wood from the mpingo tree."
    "In 1971, astronaut Alan Shepard played golf on the moon."
    "Or did he?"
)
``` 
but this does:
```Python
facts = (
    "Clarinets are made almost entirely out of wood from the mpingo tree.",
    "In 1971, astronaut Alan Shepard played golf on the moon."
    "Or did he?"
)
``` 
So I guess the logic of ISC004 requires at least two elements, to show that it is a collection (assuming the comma isn't unintentional :rofl:) - fair enough.

But shouldn't the same logic apply to the wrapped 3-liner at the top: if there is only one element, it is probably not a (nested) collection regardless of how many string lines there are.

That's still imperfect, but maybe the best that can be done given the use of parentheses by Python for both grouping and tuples?

**Comment by hugovk** [comment]
Thanks for the report, this looks like a bug when there's three or more strings.

With Ruff, the first example doesn't fail (and the second and third have the same behaviour as here).

I've opened #83 to fix this and another discrepancy with Ruff.

cc @hauntsaninja who did the Ruff version of ISC004.
